### PR TITLE
refactor(pick-list, pick-list-group, pick-list-item): Design specs for pick list matched

### PR DESCRIPTION
**Pull request by rmstinson**
**Related Issue:** #2454

## Summary

Matched the visual specs for pick list from Figma, one behavioral issue pending. Hover state is an issue on the action slot. See below:

Picklist Behavior

### CODE REVIEW

**Review by caripizza — CHANGES_REQUESTED** [review]
This is looking good @rmstinson, but can you clean it up a bit - remove whitespace changes and commented out code?

**Comment by jcfranco** [comment]
> remove whitespace changes

Formatting should be taken care of automatically when committing. Can you run `npm install && npm run util:prep-autolint`?

**Comment by bstifle** [comment]
@rmstinson looks like a couple visual bugs were introduced on hover:
image

image

image

hovering the x/end action should only hover the X hit area bg:
image
image

**Comment by rmstinson** [comment]
@bstifle Yep, the hover bug was mentioned up top. Since the pick-list is full-width, and the action buttons are nested in the pick-list, there doesn't seem to be a way to prevent the hover state from being incorrectly triggered. I think that the host element needs to get reworked in order to get the hover state to work correctly. (this applies to the Value List too, which leverages the Pick List as like most of itself.) Syncing with @asangma tomorrow to address.

**Comment by jcfranco** [comment]
Moving to the next sprint.

**Review by rmstinson — COMMENTED** [review]
Made requested changes

**Comment by bstifle** [comment]
i have a finesse ask.

this makes sense, because the active indicator is separated by the action:
image

Can we make some logic to where if no action-start, this does not happen?:
image

would make more sense if the active indicator and item hit areas were married in this case

**Comment by bstifle** [comment]
that being said.... does the active indicator area even need its own affordance? its acting like an invisible checkbox

**Comment by rmstinson** [comment]
@bstifle Good point, what you're describing is the desired behavior once we refactor the component itself and not just the styling. Because of the way the host is built, the current hover behavior is a little inaccurate to the designs. The icon slot and the label slots can't have active hover states at the same time, without also being active when the right-action slot is hovered on.

action_hoverstate

Once we're able to refactor the actual architecture of the component, it'll be super-duper easy to fix.

pick_hoverstate

**Comment by bstifle** [comment]
dope dope dope

**Comment by bstifle** [comment]
@rmstinson hey for top and bottom padding... can we get those back to p2 (8px)

Alan and I had worked on those to get them a little tighter

**Comment by rmstinson** [comment]
@bstifle Changes made! 
Screen Shot 2021-07-20 at 3 47 05 PM

**Review by bstifle — APPROVED** [review]
Lookin good mr. mike

**Comment by julio8a** [comment]
@rmstinson, can you merge master into this?

**Review by asangma — CHANGES_REQUESTED** [review]
A couple minor things. 
Looks good otherwise!

**Review by rmstinson — COMMENTED** [review]
Question about min-height attribute for Header-only, no right-action pick list

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss [hunk 1]
```diff
@@ -1,74 +1,84 @@
...
     flex
     text-color-1
     shadow-border-bottom
     m-0
     mb-px;
-  transition: background-color $transition;
-  animation: calcite-fade-in $transition;
+	transition: background-color $transition;
+	animation: calcite-fade-in $transition;
+	min-height: 3rem;
```
[the remark this answers is not in the document]
Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.scss:11]: It's not ideal, but it should be left up to the user and the content.
In most cases, a list should consist of homogenous content.

**Comment by bstifle** [comment]
Looks good Mike, let's ship this to the moon

**Comment by julio8a** [comment]
We need a code review

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss [hunk 2]
```diff
@@ -1,74 +1,81 @@
 :host {
-  @extend %component-host;
-  @apply items-stretch
+	@extend %component-host;
```
Comment (asangma) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.scss:2]: We should probably replace this with any relevant styles from this partial.

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss [hunk 3]
```diff
@@ -1,74 +1,81 @@
...
+	@apply items-stretch
     flex
     text-color-1
     shadow-border-bottom
     m-0
     mb-px;
-  transition: background-color $transition;
-  animation: calcite-fade-in $transition;
+	transition: background-color $transition;
```
Comment (asangma) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.scss:9]: Are we able to use values from the Tailwind config here?

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss [hunk 4]
```diff
@@ -1,74 +1,81 @@
...
+	animation: calcite-fade-in $transition;
```
Comment (asangma) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.scss:10]: Are we able to use values from the Tailwind config here?

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss [hunk 5]
```diff
@@ -1,74 +1,81 @@
...
 }
 
-.label {
-  @apply bg-transparent flex flex-auto px-3 py-2 items-center cursor-pointer focus-base;
-  &:focus {
-    @apply focus-inset;
-  }
+:host([selected]) .icon-dot:before,
+:host([selected]) .icon calcite-icon {
+	transition: opacity $transition;
```
Comment (asangma) [inline src/components/calcite-pick-list-item/calcite-pick-list-item.scss:47]: Are we able to use values from the Tailwind config here?

**Review by asangma — DISMISSED** [review]
One change request.

The Sass var comments might need to be handled in a separate PR if a value needs to be added to the Tailwind config.

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss (hunk 2 above)
Comment (asangma) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.scss:2]: Host partial is [here](https://github.com/Esri/calcite-components/blob/master/src/assets/styles/_host.scss).

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss (hunk 3 above)
Comment (rmstinson) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.scss:9]: For the transition?

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss (hunk 2 above)
Comment (rmstinson) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.scss:2]: Handling in a separate PR.

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss (hunk 3 above)
Comment (rmstinson) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.scss:9]: Handling in a separate PR.

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss (hunk 5 above)
Comment (rmstinson) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.scss:47]: Handling in a separate PR.

File: src/components/calcite-pick-list-item/calcite-pick-list-item.scss (hunk 4 above)
Comment (rmstinson) [reply src/components/calcite-pick-list-item/calcite-pick-list-item.scss:10]: Handling in a separate PR.

**Pull request merged**